components/skeleton/test/skeleton-group.vdiff.js

```diff
@@ -0,0 +1,235 @@
+		const elem = await fixture(noSkeletonFixture);
+		elem.insertAdjacentHTML('beforeend', '<d2l-test-skeleton-container skeleton></d2l-test-skeleton-container>');
+		await elem.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	it('remove-element', async() => {
+		const elem = await fixture(oneSkeletonFixture);
+		elem.querySelector('d2l-test-skeleton-container[skeleton]').remove();
+		await elem.updateComplete;
+		await expect(elem).to.be.golden();
+	});
+
+	[true, false].forEach((skeleton) => {
+		it(`mixed-elements${skeleton ? '-skeleton' : ''}`, async() => {
+			const elem = await fixture(html`
+				<d2l-skeleton-group-test-wrapper style="width: 330px">
+					<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+					<d2l-test-skeleton-container class="to-skeleton"></d2l-test-skeleton-container>
+					<d2l-test-skeleton-box></d2l-test-skeleton-box>
+				</d2l-skeleton-group-test-wrapper>
+			`);
+			elem.querySelector('.to-skeleton').skeleton = skeleton;
+			await expect(elem).to.be.golden();
+		});
+	});
+
+	[
+		{
+			name: 'nested-groups-inner',
+			template: html`
+				<d2l-skeleton-group-test-wrapper style="width: 330px">
+					<d2l-skeleton-group-test-wrapper>
+						<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+						<d2l-test-skeleton-box></d2l-test-skeleton-box>
+						<d2l-test-skeleton-container class="skeleton-element"></d2l-test-skeleton-container>
+					</d2l-skeleton-group-test-wrapper>
+					<d2l-skeleton-group-test-wrapper>
+						<d2l-test-skeleton-box></d2l-test-skeleton-box>
+					</d2l-skeleton-group-test-wrapper>
+				</d2l-skeleton-group-test-wrapper>
+			`
+		},
+		{
+			name: 'nested-groups-middle',
+			template: html`
+				<d2l-skeleton-group-test-wrapper style="width: 330px">
+					<d2l-skeleton-group-test-wrapper class="skeleton-element">
+						<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+						<d2l-test-skeleton-box></d2l-test-skeleton-box>
+						<d2l-test-skeleton-container></d2l-test-skeleton-container>
+					</d2l-skeleton-group-test-wrapper>
+					<d2l-skeleton-group-test-wrapper>
+						<d2l-test-skeleton-box></d2l-test-skeleton-box>
+					</d2l-skeleton-group-test-wrapper>
+				</d2l-skeleton-group-test-wrapper>
+			`
+		},
+		{
+			name: 'nested-groups-outer',
+			template: html`
+				<div style="width: 330px">
+					<d2l-skeleton-group-test-wrapper class="skeleton-element">
+						<d2l-skeleton-group-test-wrapper>
+							<d2l-test-skeleton-heading level="1">Heading 1</d2l-test-skeleton-heading>
+							<d2l-test-skeleton-box></d2l-test-skeleton-box>
+							<d2l-test-skeleton-container></d2l-test-skeleton-container>
+						</d2l-skeleton-group-test-wrapper>
+						<d2l-skeleton-group-test-wrapper>
+							<d2l-test-skeleton-box></d2l-test-skeleton-box>
+						</d2l-skeleton-group-test-wrapper>
+					</d2l-skeleton-group-test-wrapper>
+				</div>
+			`
+		}
+	].forEach(({ name, template }) => {
```

Comment:
wow, this is a great way to have tests with the same logic and different fixtures 😮